writer.go

```diff
@@ -763,6 +763,11 @@ func (c *writerColumn) flushFilterPages() error {
 			c.page.filter = c.newBloomFilterEncoder(numValues)
 		}
 		for _, page := range c.filter {
+			dict := page.Dictionary()
+			if dict != nil {
+				dict.WriteTo(c.page.filter)
+				continue
+			}
```

Comment:
It should be safe to assume that if one page has a dictionary, all pages do and will share the dictionary since they are part of the same column chunk here? 

Could we lookup the dictionary once and have two loops instead?